data-prepper-plugins/in-memory-source-coordination-store/src/test/java/org/opensearch/dataprepper/plugins/sourcecoordinator/inmemory/InMemoryPartitionAccessorTest.java

```diff
@@ -0,0 +1,157 @@
+        given(item.getSourcePartitionKey()).willReturn(partitionKey);
+        given(item.getSourcePartitionStatus()).willReturn(SourcePartitionStatus.CLOSED);
+        given(item.getReOpenAt()).willReturn(Instant.now().plusSeconds(10))
+                .willReturn(Instant.now().plusSeconds(10))
+                .willReturn(Instant.now().minusSeconds(10))
+                .willReturn(Instant.now().minusSeconds(10));
+
+        final InMemorySourcePartitionStoreItem secondItem = mock(InMemorySourcePartitionStoreItem.class);
+        given(secondItem.getSourcePartitionKey()).willReturn(secondPartitionKey);
+
+        objectUnderTest.queuePartition(item);
+        objectUnderTest.queuePartition(secondItem);
+
+        final Optional<SourcePartitionStoreItem> resultItem = objectUnderTest.getNextItem();
+
+        assertThat(resultItem.isPresent(), equalTo(true));
+        assertThat(resultItem.get(), equalTo(secondItem));
+
+        final Optional<SourcePartitionStoreItem> secondResultItem = objectUnderTest.getNextItem();
+
+        assertThat(secondResultItem.isPresent(), equalTo(true));
+        assertThat(secondResultItem.get(), equalTo(item));
+    }
+
+    @Test
+    void updateItem_that_does_not_exist_does_not_update_or_queue_item() {
+        final String partitionKey = UUID.randomUUID().toString();
+
+        final InMemorySourcePartitionStoreItem updateItem = mock(InMemorySourcePartitionStoreItem.class);
+        given(updateItem.getSourcePartitionKey()).willReturn(partitionKey);
+
+        objectUnderTest.updateItem(updateItem);
+
+        final Optional<SourcePartitionStoreItem> emptyItem = objectUnderTest.getNextItem();
+        assertThat(emptyItem.isEmpty(), equalTo(true));
+
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"CLOSED", "UNASSIGNED"})
+    void updateItem_with_unassigned_or_closed_status_requeues_the_item(final String status) {
+        final String partitionKey = UUID.randomUUID().toString();
+
+        final InMemorySourcePartitionStoreItem item = mock(InMemorySourcePartitionStoreItem.class);
+        given(item.getSourcePartitionKey()).willReturn(partitionKey);
+
+        objectUnderTest.queuePartition(item);
+
+        final Optional<SourcePartitionStoreItem> originalItem = objectUnderTest.getNextItem();
+
+        assertThat(originalItem.isPresent(), equalTo(true));
+        assertThat(originalItem.get(), equalTo(item));
+
+        final InMemorySourcePartitionStoreItem updateItem = mock(InMemorySourcePartitionStoreItem.class);
+        given(updateItem.getSourcePartitionKey()).willReturn(partitionKey);
+        given(updateItem.getSourcePartitionStatus()).willReturn(SourcePartitionStatus.valueOf(status));
+
+        objectUnderTest.updateItem(updateItem);
+
+        final Optional<SourcePartitionStoreItem> updatedItem = objectUnderTest.getNextItem();
+        assertThat(updatedItem.isPresent(), equalTo(true));
+        assertThat(updatedItem.get(), equalTo(updateItem));
+    }
+
```

Comment:
You will need a test case with multiple threads. In memory Source coordination is between multiple threads on a single node, right?

Reply:
No it's really just for one thread with on node